src/Shared/FrameworkLocationHelper.cs

```diff
@@ -622,6 +623,25 @@ private static string PathToV4ToolsInFallbackDotNetFrameworkSdk
             }
         }
 
+        private static string pathOfInstalledWindowsKits = string.Empty;
+
+        internal static string PathOfInstalledWindowsKits
+        {
+            get
+            {
+                #if FEATURE_WIN32_REGISTRY
+                if (string.IsNullOrEmpty(FrameworkLocationHelper.pathOfInstalledWindowsKits))
+                {
+                    FrameworkLocationHelper.pathOfInstalledWindowsKits = FindRegistryValueUnderKey(
```

Comment:
I am kind of worried that we don't provide any fallback path here, because FindRegistryValueUnderKey may return null.
@rainersigwald , does it make sense?